store/local/job_spec_adapter.go

```diff
@@ -568,8 +530,19 @@ func (JobSpecAdapter) FromSpec(spec models.JobSpec) (Job, error) {
 		})
 	}
 
+	var truncateTo, offset, size string
+	if spec.Task.Window != nil {
```

Comment:
why we need this check? and these intermediate variables we can directly refer to the window variables

Reply:
this is to avoid panic. for example, the spec being used does not have `Window` field being configured, meaning it's `nil`. if we do not check it first and access the field `Window`, panic will hapen